subcommand/controller/command.go

```diff
@@ -0,0 +1,127 @@
+package controller
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"sync"
+
+	"github.com/hashicorp/consul-k8s/api/v1alpha1"
+	"github.com/hashicorp/consul-k8s/controllers"
+	"github.com/hashicorp/consul-k8s/subcommand/flags"
+	"github.com/mitchellh/cli"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+)
+
+type Command struct {
+	UI        cli.Ui
+	flagSet   *flag.FlagSet
+	k8s       *flags.K8SFlags
+	httpFlags *flags.HTTPFlags
+
+	flagMetricsAddr          string
+	flagEnableLeaderElection bool
+
+	once sync.Once
+	help string
+}
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(v1alpha1.AddToScheme(scheme))
+	// +kubebuilder:scaffold:scheme
+}
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagMetricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
+	c.flagSet.BoolVar(&c.flagEnableLeaderElection, "enable-leader-election", false,
+		"Enable leader election for controller. "+
+			"Enabling this will ensure there is only one active controller manager.")
+	c.httpFlags = &flags.HTTPFlags{}
+	flags.Merge(c.flagSet, c.httpFlags.Flags())
+	c.help = flags.Usage(help, c.flagSet)
+}
+
+func (c *Command) Run(_ []string) int {
+	c.once.Do(c.init)
+	if err := c.flagSet.Parse(nil); err != nil {
+		setupLog.Error(err, "parsing flagSet")
+		return 1
+	}
+	if len(c.flagSet.Args()) > 0 {
+		c.UI.Error(fmt.Sprintf("Should have no non-flag arguments."))
```

Comment:
can use `setupLog` here since it's used everywhere else. Then I don't think we need to inject `UI`?

Reply:
how does that UI injection work? I have seen that in every other subcommand and im not sure if we should keep that pattern or diverge. Can revert to using the logging we have used all over